### PR TITLE
Laboratory work #4, Sviridov Vyacheslav - 20FPL1

### PR DESCRIPTION
**Pull request by VSvir**

### CODE REVIEW

**Comment by WhiteJaeger** [comment]
It is time to push changes for the fourth lab.

**Review by WhiteJaeger — COMMENTED** [review]
Keep going.

File: lab_4/main.py [hunk 1]
```diff
@@ -66,41 +107,108 @@ class NGramTextGenerator:
...
     """
-    pass
+    if not isinstance(decoded_corpus, tuple) or not decoded_corpus:
+        return ''
+    sentence = []
+    for word in decoded_corpus:
+        sentence.extend(word)
+    if sentence[0] == '_':
+        sentence[0] = ''
+    if sentence[-1] == '_':
```
Comment (WhiteJaeger) [inline lab_4/main.py:207]: Could be combined into `if-elif`

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_4/main.py (hunk 1 above)
Comment (VSvir) [reply lab_4/main.py:207]: We cannot combine these because we need to check both the beginning and the end of a sentence

**Pull request closed without merging**